### ISSUE TITLE
Transfer from deletes all cloudinary assets

### ISSUE THREAD

**Issue by ststaynov** [strapi/strapi#17530]
### Required System information

- Node.js version: v16.13.0
- NPM version: 8.4.0
- Strapi version: 4.12
- Database: PostgreSQL
- Operating system: Mac OS Ventura 13.4.1
- Is your project Javascript or Typescript: Javascript

### Describe the bug

Performing a `yarn strapi transfer --from` results in all common image assets between the local environment and the remote to be deleted from cloudinary. 

### Steps to reproduce the behavior

1. Execute `yarn strapi transfer --from ${remote heroku server}/admin` from the terminal
2. Action completes successfully
3. All commonly shared image assets between the local environment and the production are deleted from the cloudinary provider.

### Expected behavior

No assets are deleted and images are transferred.

### Additional

Links to assets are correctly transferred, however they are not found as they are deleted from cloudinary.
Production and local environment use the same cloudinary distribution.

**Comment by derrickmehaffy** [comment]
@marcoautiero I'm fairly certain this is intended functionality right? We provide a warning that all data and assets on the destination will be deleted until we add the incremental or alternative sync strat?

**Comment by ststaynov** [comment]
I do agree that this is an logical chain of events, to first delete the old images and then upload the new ones, however in the case that the same cludinary distribution is used I would expect for the common images to not be affected. Currently the images are deleted and then there are none to be transfered as they are not found.

**Comment by derrickmehaffy** [comment]
> I do agree that this is an logical chain of events, to first delete the old images and then upload the new ones, however in the case that the same cludinary distribution is used I would expect for the common images to not be affected. Currently the images are deleted and then there are none to be transfered as they are not found.

Ahhhh I see, it's because the same cloudinary instance is being used. I'm like 99% sure we didn't account for that as it's not recommend. The source and destination don't communicate to compare their configuration, the source just sends the information and the destination just places it where it should.

Cc @marcoautiero it was neither of what we talked about yesterday lol 😆

**Comment by ststaynov** [comment]
Agree, awful practice to use the same cloudinary instance. When time is not on your side, however, it can prove handy. The handy part is abolished by finding all your images deleted and especially if you don't have a cloudinary backup (luckily I did 🙂). 
I still think this will happen to several other folks (perhaps some without backups) but would also understand if you don't want to cover this case.

**Comment by jbstewart** [comment]
This just happened to me. I think at a minimum the documentation for using the Cloudinary plugin (which I can't find right now, so maybe it doesn't exist) should specify that you need to have a different Cloudinary environment for each instance, local, production, staging, etc. Cloudinary does not allow but one environment on their free plan, but on their site they do recommend that you can create multiple free accounts for this purpose.

**Comment by iamterryclark** [comment]
This just happened tome now also!

So annoying!!

it just bulk deleted everything i was transferring from my online instance to my offline instance to update some things and this happened, need to speak with cloudinary now to rerstore.

**Comment by acropup** [comment]
I fell for this trap as well. Fortunately, I was able to speak with Cloudinary support and get the files recovered, even though I had never turned on automatic backup. They provided a [CSV template](http://res.cloudinary.com/demo/raw/upload/recover_sample.csv) for which I was to list all the files that needed recovering. I queried `https://<Strapi_URL>/api/upload/files` and used the media resource URLs within the response to extract the public IDs for each of the deleted files. I eventually managed a full recovery, but this cost me a lot of stress and a couple day's work, and we're now paying $100/mo to Cloudinary for extended backups and a second account for our staging environment. I'm glad that this wasn't a total disaster, but it was not good.

@derrickmehaffy:
>@marcoautiero I'm fairly certain this is intended functionality right? We provide a warning that all data and assets on the destination will be deleted until we add the incremental or alternative sync strat?

This may very well be intended functionality, but I would argue that the warning is insufficiently clear. The current warning is:

_The transfer will delete all the **local Strapi assets** and its **database**. Are you sure you want to proceed? Yes_

Nothing in the warning says to me that **files on an entirely different cloud host** will also be deleted. A sufficient warning would need to be something more like:

_The transfer will delete all the local Strapi assets and its database. Please note that if you use the Media Library with an upload provider or CDN (such as Cloudinary or Amazon S3), the `strapi transfer` command also instructs the CDN to delete all static content referenced within the Strapi Media Library (this includes images, videos, documents, and other downloadable files). Furthermore, if the `--to` and `--from` Strapi instances are already mirror images of each other, with identical upload provider configurations, one more `transfer` command will delete the one and only source of static files from the CDN that both of the Strapi Media Libraries are using. This will happen even if the `--exclude files` or `--only content` flags are used. This is because files remotely stored on a CDN are considered "content" in the context of a `strapi transfer`, because it is a side-effect of clearing the local Strapi Media Library database that all remote media files are deleted._

I don't know what happens if files are stored locally on the file system, without an upload provider. I also don't know if there are other upload providers for which this is just as serious a problem (imgix, bunny.net?).

@ststaynov:
> Agree, awful practice to use the same cloudinary instance. [...]

I agree that it's an awful practice when the tools you use make it so dangerous, but I see nothing inherently wrong with the desire to work this way. I would like to clone the state of my CMS without copying gigabytes of static media files for no reason but to avoid catastrophe. If there's an alternative strategy that I'm not seeing--something that can be done safely and doesn't take hours to complete or result in cloud overage charges--I'd love a more detailed explanation of what good practice here looks like!

**Comment by cache-your-dreams** [comment]
@dosu
